backend/tarsy/services/history_service.py

```
@@ -356,6 +356,26 @@ def _get_session_with_stages_operation():
         
         result = self._retry_database_operation("get_session_with_stages", _get_session_with_stages_operation)
         return result
+
+    def get_stage_interaction_counts(self, execution_ids: List[str]) -> Dict[str, Dict[str, int]]:
```

Ruff (0.12.2)

Line too long (98 > 88)

(E501)